test/address.spec.ts

```
@@ -1,782 +1,661 @@
+          }
+        });
+
+        it('should return near metric coordinates when radius is undefined', () => {
+          const latitude = parseFloat(faker.address.latitude());
+          const longitude = parseFloat(faker.address.longitude());
+          const isMetric = true;
+
+          const coordinate = faker.address.nearbyGPSCoordinate(
+            [latitude, longitude],
+            undefined,
+            isMetric
+          );
+
+          expect(coordinate.length).toBe(2);
+          expect(typeof coordinate[0]).toBe('string');
+          expect(typeof coordinate[1]).toBe('string');
+
+          const distanceToTarget =
+            Math.pow(coordinate[0] - latitude, 2) +
+            Math.pow(coordinate[1] - longitude, 2);
+
+          console.log([latitude, longitude], coordinate, distanceToTarget);
+
+          expect(distanceToTarget).lessThanOrEqual(
+            100 * 0.002 // 100 km ~= 0.9 degrees, we take 2 degrees
+          );
+        });
+
+        it('should return near non metric coordinates when radius is undefined', () => {
+          const latitude = parseFloat(faker.address.latitude());
+          const longitude = parseFloat(faker.address.longitude());
+          const isMetric = false;
+
+          const coordinate = faker.address.nearbyGPSCoordinate(
+            [latitude, longitude],
+            undefined,
+            isMetric
+          );
+
+          expect(coordinate.length).toBe(2);
+          expect(typeof coordinate[0]).toBe('string');
+          expect(typeof coordinate[1]).toBe('string');
+
+          const distanceToTarget =
+            Math.pow(coordinate[0] - latitude, 2) +
+            Math.pow(coordinate[1] - longitude, 2);
+
+          console.log([latitude, longitude], coordinate, distanceToTarget);
+
+          expect(distanceToTarget).lessThanOrEqual(
+            100 * 0.002 * 1.6093444978925633 // 100 miles to km ~= 0.9 degrees, we take 2 degrees
+          );
```

This is currently failing hard...
I don't know if the function is wrong or if my test is wrong.

2 degrees are like 200+ km and the factor 1.6 converts the 100km to miles
also `isMetric = false` should then use the miles version of the function
but the delta is very huge :thinking:

---

@prisis what do you think I should do here?
Currently it blocks the PR to get merged, so maybe we should mark it with `it.todo` and look into that later? Maybe even create an issue to track it?
The benefit would be, that we would have now a much better test coverage for the rest of the address module.

---

Mark it as todo for now and maybe add an issue so it doesn't get lost.

---

https://github.com/faker-js/faker/issues/321